api.js

```diff
@@ -43,12 +45,12 @@ exports.init = function (sbot, config) {
   }
 
   function find(metafeed, visit, maybeCB) {
+    const cb = maybeCB
```

Comment:
Yeah it seems the logic for cb here is all upside down. It's supposed to be like `create()` (line 70) because the arguments are flexible and we don't know if we have a callback. I think `find` and `filter` used to be public APIs but now that they're private, we can perhaps assume that the arguments are non-flexible (i.e. cb is always given, and always in the 3rd slot).

Note however that this is completely unrelated to your PR. Don't work on refactoring this in this PR.

Reply:
I had a test fail because of this, so I fixed it.
I would like some tooling that catches broken code like this (my linting/highlighting sees it in my editor)

Reply:
This is an example of something I'm not going to revert / split out.
It's not a good use of time and the code was broken as it was

Reply:
I wasn't asking you to do a refactor, I was just suggesting about a new refactor, i.e.

> we can perhaps assume that the arguments are non-flexible

but I was advising to not do this new suggestion in this PR.